bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleRegistryImpl.java

```diff
@@ -251,7 +255,7 @@ public Rule add(Rule rule) {
      * @param rule candidate for unique ID
      * @return a rule with UID
      */
-    protected Rule initRuleId(String rUID, Rule rule) {
+    protected @Nonnull Rule initRuleId(String rUID, Rule rule) {
```

Comment:
OK I agree that it is more readable way, but it adds additional checks at runtime, which are not needed for me. Actually this is not public method used only by the RuleRegistryImpl and RuleEngine. Anyway I can live with it.